src/sdl2_compat.c

```diff
@@ -1905,6 +1907,80 @@ SDL_WarpMouseGlobal(int x, int y)
     return SDL3_WarpMouseGlobal((float)x, (float)y);
 }
 
+/* The SDL2 version of SDL_Surface changed, so we need to convert when necessary. */
+
+
+struct SDL2_Surface
+{
+    Uint32 flags;               /**< Read-only */
+    SDL_PixelFormat *format;    /**< Read-only */
+    int w, h;                   /**< Read-only */
+    int pitch;                  /**< Read-only */
+    void *pixels;               /**< Read-write */
+
+    /** Application data associated with the surface */
+    void *userdata;             /**< Read-write */
+
+    /** information needed for surfaces requiring locks */
+    int locked;                 /**< Read-only */
+
+    /** list of BlitMap that hold a reference to this surface */
+    void *list_blitmap;         /**< Private */
+
+    /** clipping information */
+    SDL_Rect clip_rect;         /**< Read-only */
+
+    /** info for fast blit mapping to other surfaces */
+    SDL_BlitMap *map;           /**< Private */
+
+    /** Reference count -- used when freeing surface */
+    int refcount;               /**< Read-mostly */
+};
+
+void Surface2to3_init(SDL2_Surface *s2, SDL_Surface *s3)
+{
+    if (!s2 || !s3) {
+        return;
+    }
+
+    s3->flags = s2->flags;
+    s3->format = s2->format;
+    s3->w = s2->w;
+    s3->h = s2->h;
+    s3->pitch = s2->pitch;
+    s3->pixels = s2->pixels;
+    // ignore user data?
+    s3->locked = s2->locked;
+    s3->list_blitmap = s2->list_blitmap;
+    s3->clip_rect = s2->clip_rect;
+    s3->map = s2->map;
+    s3->refcount = s2->refcount;
+}
+
+void Surface3to2_init(SDL_Surface *s3, SDL2_Surface *s2)
+{
+    if (!s2 || !s3) {
+        return;
+    }
+    s2->flags = s3->flags;
+    s2->format = s3->format;
+    s2->w = s3->w;
+    s2->h = s3->h;
+    s2->pitch = s3->pitch;
+    s2->pixels = s3->pixels;
+    // ignore user data?
```

Comment:
... as above

Reply:
probably also ok. we don't want to overwrite the user data.
we just want to fetch back the sdl3 surface fields.